### PR TITLE
Ngfw 14300 increase tcp receive buffer size

### PR DESCRIPTION
**Pull request by cblaise-untangle**

### CODE REVIEW

File: uvm/impl/com/untangle/uvm/AppTCPSessionImpl.java [hunk 1]
```diff
@@ -37,6 +41,23 @@ public class AppTCPSessionImpl extends AppSessionImpl implements AppTCPSession
     protected boolean[] lineBuffering = new boolean[] { false, false };
     protected ByteBuffer[] readBuf = new ByteBuffer[] { null, null };
 
+    /**
+     * Size of buffer to work with.
+     * The faster the network and ability to process, the bigger the size performs throughput impact
+     */
+    private static int Buffer_size = 8 * 1024;
```
Comment (acooke-arista) [inline uvm/impl/com/untangle/uvm/AppTCPSessionImpl.java:48]: This same value is used in TCPSource.java. Could it be global?

File: uvm/hier/usr/lib/python3/dist-packages/tests/global_functions.py [hunk 2]
```diff
@@ -143,18 +141,251 @@ def verify_iperf_configuration(wan_ip):
...
+
+    last_line = iperf_results.splitlines()[-1]
+    matches = iperf_results_re.match(last_line)
+    transferred_value = 0
+    rate_value = 0
+    if matches is not None:
+        # If we don't get here either:
+        # - Something fundamentally wrong went wrong with the iperf command
+        # - Our regex failed
+        start_time = matches.group(1)
```
Comment (acooke-arista) [inline uvm/hier/usr/lib/python3/dist-packages/tests/global_functions.py:185]: Some variables are assigned but not used, are they necessary?

File: uvm/hier/usr/lib/python3/dist-packages/tests/test_peformance.py [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+        ##
+        iperf_running = 0
+        timeout = 60
+        while iperf_running == 0 and timeout > 0:
+            timeout -= 1
+            remote_control.run_command("pkill --signal 9 iperf", host=global_functions.iperf_server)
+            time.sleep(1)
+            iperf_running = remote_control.run_command("pidof iperf", host=global_functions.iperf_server)
+
+        assert(True is True)
```
Comment (acooke-arista) [inline uvm/hier/usr/lib/python3/dist-packages/tests/test_peformance.py:225]: I assume this is here for figuring things out and should probably be removed?

File: uvm/impl/com/untangle/uvm/AppTCPSessionImpl.java (hunk 1 above)
Comment (cblaise-untangle) [reply uvm/impl/com/untangle/uvm/AppTCPSessionImpl.java:48]: Not easily, but the point of specifying the tcp_buffer_size value from the command line essentially makes it global.

File: uvm/hier/usr/lib/python3/dist-packages/tests/global_functions.py (hunk 2 above)
Comment (cblaise-untangle) [reply uvm/hier/usr/lib/python3/dist-packages/tests/global_functions.py:185]: Good point.  I should include those values as a "duration" field.

File: uvm/hier/usr/lib/python3/dist-packages/tests/test_peformance.py (hunk 3 above)
Comment (cblaise-untangle) [reply uvm/hier/usr/lib/python3/dist-packages/tests/test_peformance.py:225]: Thanks for catching that.

File: uvm/hier/usr/lib/python3/dist-packages/tests/test_peformance.py [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+
+        # Skip if iperf not installed
+        iperf_avail = global_functions.verify_iperf_configuration(wan_ip, ignore_running=True)
+        if (not iperf_avail):
+            raise unittest.SkipTest("iperf_server test client unreachable")
+
+        # Start iperf server
+        iperf_server_running = remote_control.run_command("pidof iperf", host=global_functions.iperf_server)
+        if iperf_server_running == 1:
+            ## !!! change to use build_iperf_command
```
Comment (acooke-arista) [inline uvm/hier/usr/lib/python3/dist-packages/tests/test_peformance.py:66]: This looks like a command made while developing, should probably be removed

File: uvm/impl/com/untangle/uvm/AppTCPSessionImpl.java (hunk 1 above)
Comment (acooke-arista) [reply uvm/impl/com/untangle/uvm/AppTCPSessionImpl.java:48]: Makes sense to me! Resolving

**Review by acooke-arista — APPROVED** [review]

**Review by acooke-arista — APPROVED** [review]

**Pull request merged**